extensions/kubernetes-client/deployment/src/main/java-templates/io/quarkus/kubernetes/client/deployment/Versions.java

```diff
@@ -0,0 +1,8 @@
+package io.quarkus.kubernetes.client.deployment;
+
+public class Versions {
+
+  public static final String QUARKUS = "${project.version}";
+  public static final String KUBERNETES_CLIENT = "${kubernetes-client.version}";
```

Comment:
Wouldn't it make more sense to add a class in the Kubernetes Client? Because here you have the version with which Quarkus was built, but not the actual runtime version (which could be overloaded locally).

As for the Quarkus version, it's probably something we should have in the core artifact. I wonder if there's a build item available already given we can display the version at startup. Maybe @geoand or @aloubyansky would know.

Reply:
Right, you will probably want to check the runtime version of those. You could use `ApplicationModel.getDependencies()` for that. We could surely add a simple utility method to return the core version.

Reply:
Currently the Quarkus version is only available at build time via https://github.com/quarkusio/quarkus/blob/2.15.1.Final/core/builder/src/main/java/io/quarkus/builder/Version.java

Reply:
> Wouldn't it make more sense to add a class in the Kubernetes Client?

It is already available in the `kubernetes-client`:
https://github.com/fabric8io/kubernetes-client/blob/677884ca88f3911f9b41ec919db152d441ad2cdd/kubernetes-client-api/src/main/java-templates/io/fabric8/kubernetes/client/Version.java

> Because here you have the version with which Quarkus was built, but not the actual runtime version

this is exactly the objective of this PR.
Sadly, we keep facing problems when a user tries to use a `kubernetes-client` version that is different from the one used at build time.

Note the `kubernetes-client` versions to compare here are 3:
- Java Operator SDK transitive dependency
- Quarkus build time
- User runtime

The proposed solution here is to:
- generate a java source file with the build time versions of the libraries at build time
- cross-compare the build time versions AND the runtime versions in a build step in the Quarkus Operator SDK
- if something is not aligned we can break the build or, at least, emit a warning

Reply:
> Sadly, we keep facing problems when a user tries to use a kubernetes-client version that is different from the one used at build time.

The build time of the Quarkus itself or the build time of the application?

Reply:
@aloubyansky Quarkus itself, and/or the version used in the Java Operator SDK

Reply:
As @andreaTP mentioned, this work is meant to help enforce that versions of the kubernetes client are aligned between all the parts. This has also been (somewhat tangentially) discussed in https://quarkusio.zulipchat.com/#narrow/stream/187038-dev/topic/quarkus.2Eplatform.2Eversion.20issue.3F